client/api/notebook.py

```diff
@@ -21,6 +23,8 @@ def run(self, protocol, messages, **kwargs):
         return self.assignment.protocol_map[protocol].run(messages, **kwargs)
 
     def auth(self, force=False, inline=True):
+        if not force and login_with_env(self.assignment):
```

Comment:
Do you think this should be moved to assignment.authenticate? I put it here since it's unique to notebooks for now.